model/atmosphere/advection/src/icon4py/model/atmosphere/advection/advection_vertical.py

```diff
@@ -221,18 +542,68 @@ def __init__(
         self._end_cell_local = self._grid.end_index(cell_domain(h_grid.Zone.LOCAL))
         self._end_cell_end = self._grid.end_index(cell_domain(h_grid.Zone.END))
 
+        # fields
+        self._k_field = field_alloc.allocate_indices(
+            dims.KDim, grid=self._grid, is_halfdim=True, dtype=gtx.int32, backend=self._backend
+        )  # TODO (dastrm): should be KHalfDim
+
+        # stencils
+        self._compute_vertical_tracer_flux_upwind = (
+            compute_vertical_tracer_flux_upwind.with_backend(self._backend)
+        )
+        self._init_constant_cell_kdim_field = init_constant_cell_kdim_field.with_backend(
+            self._backend
+        )
+        self._integrate_tracer_vertically = integrate_tracer_vertically.with_backend(self._backend)
+
         log.debug("vertical advection class init - end")
 
+    def _get_horizontal_start_end(self, even_timestep: bool):
+        if even_timestep:
+            horizontal_start = self._start_cell_lateral_boundary_level_2
+            horizontal_end = self._end_cell_end
+        else:
+            horizontal_start = self._start_cell_nudging
+            horizontal_end = self._end_cell_local
```

Comment:
since `horizontal_end` is the same regardless of `even_timestep`, you can put it outside of the if statement